common-libs/user-management/src/CognitoIdentityService.ts

```diff
@@ -187,7 +195,14 @@ export class CognitoIdentityService {
 
     try {
       const result = await this.cognitoidentityserviceprovider.respondToAuthChallenge(params).promise()
-      tempSession[USERNAME] = result.Session
+      //NOTE : successful OTP return a undefined session . wrong code return a new session
+      if (result.Session) {
+        tempSession[hashedTokenSession] = result.Session
+      } else {
+        //TODO : we still need to think about clean up for sessions that was not finished by user. ex. session was confirmed with wrong pasword 1 or 2 times with out sucess.
+        // potential memory leak.
+        delete tempSession[hashedTokenSession]
```

Comment:
If we only need to do this when OTP was successful, then probably the better place to do this is right before returning `CompleteLoginPasswordlessResult.Success`